pennylane/devices/qutrit_mixed/measure.py

```diff
@@ -0,0 +1,296 @@
+        return math.reshape(summed_probs, new_shape)
+
+    return probs
+
+
+def calculate_variance(
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Find variance of observable.
+
+    Args:
+        measurementprocess (StateMeasurement): measurement to apply to the state.
+        state (TensorLike): state to apply the measurement to.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        TensorLike: the result of the measurement.
+    """
+    probs = calculate_probability(measurementprocess, state, is_state_batched)
+    eigvals = math.asarray(measurementprocess.eigvals(), dtype="float64")
+    # In case of broadcasting, `probs` has two axes and these are a matrix-vector products
+    return math.dot(probs, (eigvals**2)) - math.dot(probs, eigvals) ** 2
+
+
+# pylint: disable=too-many-return-statements
+def get_measurement_function(
+    measurementprocess: MeasurementProcess,
+) -> Callable[[MeasurementProcess, TensorLike], TensorLike]:
+    """Get the appropriate method for performing a measurement.
+
+    Args:
+        measurementprocess (MeasurementProcess): measurement process to apply to the state.
+        state (TensorLike): the state to measure.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        Callable: function that returns the measurement result.
+    """
+    if isinstance(measurementprocess, StateMeasurement):
+        if isinstance(measurementprocess, ExpectationMP):
+            # TODO add faster methods
+            # TODO add support for sparce Hamiltonians
+            if isinstance(measurementprocess.obs, (Hamiltonian, Sum)):
+                return calculate_expval_sum_of_terms
+            if measurementprocess.obs.has_matrix:
+                return calculate_expval
+        if measurementprocess.obs is None or measurementprocess.obs.has_diagonalizing_gates:
+            if isinstance(measurementprocess, StateMP):
+                return calculate_reduced_density_matrix
+            if isinstance(measurementprocess, ProbabilityMP):
+                return calculate_probability
+            if isinstance(measurementprocess, VarianceMP):
+                return calculate_variance
+
+    raise NotImplementedError
+
+
+def measure(
+    measurementprocess: MeasurementProcess, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Apply a measurement process to a state.
+
+    Args:
+        measurementprocess (MeasurementProcess): measurement process to apply to the state.
+        state (TensorLike): the state to measure.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        Tensorlike: the result of the measurement.
+    """
+    return get_measurement_function(measurementprocess)(measurementprocess, state, is_state_batched)
+
+
+def calculate_expval_sum_of_terms(
```

Comment:
Please move definition of function to before the `get_measurement_function` so that it is not forward-referenced.